File: 백준/Silver/1920. 수 찾기/수 찾기.cc
#include<iostream>
#include<algorithm>
using namespace std;
int a[100001];
int bin_search(int a[], int left, int right, int num) 
{
	while(left<=right)
	{
		int mid=(left+right)/2;	
		if(a[mid] < num) left = mid+1;
		else if(a[mid] > num) right = mid-1;
		else return 1;
	}
	return 0;
}
int main()
{
	int n,m;
	scanf(" %d", &n);
	for(int i=0;i<n;i++)
		scanf(" %d", &a[i]);
	sort(a, a+n);
	scanf(" %d", &m);
	for(int i=0;i<m;i++)
	{
		int b;
		scanf(" %d", &b);
		printf("%d\n", bin_search(a, 0, n-1, b));
		
	}
	
	
	
	return 0;
}

File: 백준/Silver/1920. 수 찾기/수 찾기.py
n = int(input())
a = list(map(int, input().split()))
a.sort()

m = int(input())
nums = list(map(int, input().split()))

def bin(l, r, t):
	while l<=r:
		mid = (l+r)//2
		if a[mid] == t:
			return 1
		elif a[mid]>t:
			r = mid-1
		elif a[mid]<t:
			l = mid+1
	return 0

for num in nums:
	print(bin(0, len(a)-1, num))